SimpleWAWS/Code/CsmExtensions/CsmSubscriptionExtensions.cs

```diff
@@ -55,11 +80,16 @@ public static partial class CsmManager
                     .Select( async r => await Load(new ResourceGroup(subscription.SubscriptionId, r.ResourceGroup.name),r.ResourceGroup, r.Resources))
                     .IgnoreAndFilterFailures();
 
-                return subscription;
+            if (deleteBadResourceGroups)
+            {
+                var deleteDuplicateResourceGroupsTasks = goodResourceGroups.Where(p => !subscription.ResourceGroups.Any(p2 => p2.CsmId == p.ResourceGroup.id))
```

Comment:
I can't figure out what this does. It seems to me that it'll never match anything. 
`subscription.ResourceGroups` == `goodResourceGroups.Load.FilterFailedLoad`

`goodResourceGroups.Where(resourceGroup not in subscription.ResourceGroups)` that condition would always be false, right? `subscription.ResourceGroups` is a subset of `goodResourceGroups`